pytest_html/plugin.py

```diff
@@ -618,15 +625,56 @@ def _save_report(self, report_content):
             with open(style_path, "w", encoding="utf-8") as f:
                 f.write(self.style_css)
 
+    def _post_process_reports(self):
+        for test_name, test_reports in self.reports.items():
+            outcome = "passed"
+            wasxfail = False
+            failure_when = None
+            full_text = ""
+            extras = []
+            duration = 0.0
+
+            # in theory the last one should have all logs so we just go
+            #  through them all to figure out the outcome, xfail, duration,
+            #    extras, and when it swapped from pass
+            for test_report in test_reports:
+                full_text += test_report.longreprtext
+                extras.extend(getattr(test_report, "extra", []))
+                duration += getattr(test_report, "duration", 0.0)
+
+                if (
+                    test_report.outcome not in ("passed", "rerun")
+                    and outcome == "passed"
+                ):
+                    outcome = test_report.outcome
+                    failure_when = test_report.when
+
+                if hasattr(test_report, "wasxfail"):
+                    wasxfail = True
+
+                if test_report.outcome == "rerun":
+                    self.append_other(test_report)
+
+            test_report.outcome = outcome
+            test_report.when = "call"
+            test_report.nodeid = test_name
+            test_report.longrepr = full_text
+            test_report.extra = extras
+            test_report.duration = duration
+
+            if wasxfail:
```

Comment:
can we simplify this to `test_report.wasxfail = wasxfail`?

Or can `test_report.wasxfail` still be `True` even if `wasxfail` is `False`?

Reply:
It can't because other code relies on the attribute existing or not. So if we do the suggested, it would be false and exist.. and lead to more changes needed than desired to go into for this changeset.

Reply:
Thanks for clarifying @csm10495, I forgot about that